pennylane/ops/pulse/convenience_functions.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This file contains convenience functions for pulse programming."""
+
+import numpy as np
+
+has_jax = True
+try:
+    import jax.numpy as jnp
+except ImportError:
+    has_jax = False
+
+
+def pwc(t):
+    """Create a function that is piecewise-constant in time, based on the params for a TDHamiltonian.
+
+    Args:
+            t(Union[float, tuple(float, float)]: the total duration as a float, or the start and end time as floats.
```

Comment:
What happens if `t` is larger than the total duration?

Reply:
Longer than the total duration of the evolution in `qml.evolve`, you mean? In that case, there are regions in `t` where the function is non-zero that are not used during the evolution. I will rephrase this to refer to non-zero regions of the function, since this is a confusing overlap with defining pulse duration in other areas of the code.

If that's not what you meant, please clarify :)

Reply:
I meant outside the time span of the `pwc`. I just saw in the changelog that it returns 0. Would be nice to explicitly say it in the docstring as well.